src/runner.ts

```diff
@@ -89,11 +89,16 @@ export class Runner {
 
     let errors = this.parser.parse(this.project);
     let prevErrors = errors;
-    let errorsExist;
+    let errorsExist: boolean;
+    let rotatedManipulators: Manipulator[];
 
     do {
       errorsExist = false;
+      rotatedManipulators = [...this.manipulators];
+
       for (const manipulator of this.manipulators) {
+        rotatedManipulators.push(rotatedManipulators.shift()!);
```

Comment:
You may want to add a little comment here about what you are doing. I took me a while to figure out that you are just rotating elements in the array FIFO. The line looks odd at first. I fact you may want to comment on the core issue you are solving.

Reply:
That's a good point, I've gone ahead and added more comments making it clearer how the Runner is working!